src/main/java/org/openstreetmap/atlas/utilities/filters/AtlasEntityPolygonsFilter.java

```diff
@@ -0,0 +1,520 @@
+                            .collect(Collectors.toSet()));
+            this.includeMultiPolygons.addAll(multiPolygons == null ? Collections.EMPTY_SET
+                    : multiPolygons.stream()
+                            .filter(notOverlappingMultipolygon(vettedPolygons, Type.INCLUDE))
+                            .collect(Collectors.toSet()));
+        }
+        else
+        {
+            this.excludePolygons.addAll(polygons == null ? Collections.EMPTY_SET
+                    : polygons.stream().filter(notOverlappingPolygon(vettedPolygons, Type.EXCLUDE))
+                            .collect(Collectors.toSet()));
+            this.excludeMultiPolygons.addAll(multiPolygons == null ? Collections.EMPTY_SET
+                    : multiPolygons.stream()
+                            .filter(notOverlappingMultipolygon(vettedPolygons, Type.EXCLUDE))
+                            .collect(Collectors.toSet()));
+        }
+    }
+
+    @Override
+    public boolean test(final AtlasEntity object)
+    {
+        return noPolygons().or(isIncluded()).or(isNotExcluded()).test(object);
+    }
+
+    private boolean excludePolygonsEmpty()
+    {
+        return this.excludePolygons.isEmpty() && this.excludeMultiPolygons.isEmpty();
+    }
+
+    private boolean includePolygonsEmpty()
+    {
+        return this.includePolygons.isEmpty() && this.includeMultiPolygons.isEmpty();
+    }
+
+    private Predicate<AtlasEntity> isIncluded()
+    {
+        return entity -> this.filterType == Type.INCLUDE && (this.includePolygons.stream().anyMatch(
+                polygon -> this.intersectionDeciding.polygonEntityIntersecting(polygon, entity))
+                || this.includeMultiPolygons.stream()
+                        .anyMatch(multiPolygon -> this.intersectionDeciding
+                                .multiPolygonEntityIntersecting(multiPolygon, entity)));
+    }
+
+    private Predicate<AtlasEntity> isNotExcluded()
+    {
+        return entity -> this.filterType == Type.EXCLUDE
+                && this.excludePolygons.stream()
+                        .noneMatch(polygon -> this.intersectionDeciding
+                                .polygonEntityIntersecting(polygon, entity))
+                && this.excludeMultiPolygons.stream()
+                        .noneMatch(multiPolygon -> this.intersectionDeciding
+                                .multiPolygonEntityIntersecting(multiPolygon, entity));
+    }
+
+    private Predicate<AtlasEntity> noPolygons()
+    {
+        return entity -> this.includePolygonsEmpty() && this.excludePolygonsEmpty();
+    }
+
+    /**
+     * Interface to be implemented if custom intersection logic is required. The
+     * {@link AtlasEntityPolygonsFilter#DEFAULT_INTERSECTION_DECIDING} is that an entity is said to
+     * intersect when any part of the entity overlaps with any area of the Polygon or MultiPolygon.
+     */
+    public interface IntersectionDeciding extends Serializable
```

Comment:
Thoughts on separating out these classes into separate files?  I think it would make this file more manageable.

Reply:
Is this useful as it's own file? Right now this seems like a very niche use. Can you think of other situations where an intersection policy would be useful?

Reply:
I think if we make a AtlasEntityPolygonsFilter sub package, we can make this file a bit more scoped.  Being a very niche use doesn't mean we can't have things in separate files or in the same file.  Separating into multiple files also manages the code structure.

Currently, this file does:
- implements the AtlasFilter (the main goal of this file)
- handles configuration interaction
- converts all different acceptable formats to be used by this filter
- defines IntersectionPolicy
- defines the filter Type
- defines the PolygonStringFormat

In my opinion, it seems like quite a bit is going on.  I think by splitting this up a bit, the file can be a bit more readable and less intimidating.  Let me know what you think!

Reply:
Under that philosophy I would say, lets move IntersectionPolicy, and the PolygonsStringFormat into their own files. Move the polygon/multipolygon String converters into the PolygonStringFormat enum. I want to keep the Type enum in the class to keep the super simple construction methods.

Reply:
I like it - if you're comfortable with it!